cogeo_mosaic/backends/stac.py

```diff
@@ -0,0 +1,151 @@
+"""cogeo-mosaic STAC backend."""
+
+import functools
+import json
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import mercantile
+import requests
+
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.utils import get_assets_from_json
+from cogeo_mosaic.mosaic import MosaicJSON
+
+
+def default_stac_accessor(feature: Dict):
+    """Return feature identifier."""
+    link = list(filter(lambda link: link["rel"] == "self", feature["links"]))
+    if link:
+        return link[0]["href"]
+
+    link = list(filter(lambda link: link["rel"] == "root", feature["links"]))
+    if link:
+        return os.path.join(
+            link[0]["href"],
+            "collections",
+            feature["collection"],
+            "items",
+            feature["id"],
+        )
+
+    # Fall back to the item ID
+    return feature["id"]
```

Comment:
@geospatial-jeff sadly this is ran at feature level so there is no real info about the stac api url. Here I choose to fallback first using the `root` and worse case to the `id` only. 

Again this will need documentation